test/db/asm/x86_64

```diff
@@ -956,6 +956,24 @@ a "dec rdi" 48ffcf
 a "dec rdx" 48ffca
 a "dec rsi" 48ffce
 a "dec WORD PTR [rdi+0x88]" 66ff8f88000000
+a "bswap rax" 480fc8
+a "bswap r15" 490fcf
+a "bswap eax" 0fc8
+a "bswap r15d" 410fcf
+ad "xchg eax, r8d" 4190
+a "xchg r8d, eax" 4190
+ad "xchg rax, rdx" 4892
+a "xchg rdx, rax" 4892
+ad "xchg rax, r8" 4990
+a "xchg r8, rax" 4990
```

Comment:
this doesnt roundtrip

Reply:
`xchg rax, rdx` and `xchg rdx, rax` have exactly the same encoding, `4892`. So either of the two cannot roundtrip. The same goes for the pair `xchg rax, r8` and `xchg r8, rax`. In general, there is an encoding for the cases where one of the operands is `ax`, `eax`, or `rax`, where the order of the operands is lost. 

Following is an excerpt from the Instruction Set Reference Manual. As you see, the encoding does not retain the order if an `ax` register is present.

<table>
<tr>
<td>90+<em>rw</em></td>
<td>XCHG AX, <em>r16</em></td>
<td>O</td>
<td>Valid</td>
<td>Valid</td>
<td>Exchange <em>r16</em> with AX.</td></tr>
<tr>
<td>90+<em>rw</em></td>
<td>XCHG <em>r16</em>, AX</td>
<td>O</td>
<td>Valid</td>
<td>Valid</td>
<td>Exchange AX with <em>r16.</em></td></tr>
<tr>
<td>90+<em>rd</em></td>
<td>XCHG EAX, <em>r32</em></td>
<td>O</td>
<td>Valid</td>
<td>Valid</td>
<td>Exchange <em>r32</em> with EAX.</td></tr>
<tr>
<td>REX.W + 90+<em>rd</em></td>
<td>XCHG RAX, <em>r64</em></td>
<td>O</td>
<td>Valid</td>
<td>N.E.</td>
<td>Exchange <em>r64</em> with RAX.</td></tr>
<tr>
<td>90+<em>rd</em></td>
<td>XCHG <em>r32</em>, EAX</td>
<td>O</td>
<td>Valid</td>
<td>Valid</td>
<td>Exchange EAX with <em>r32.</em></td></tr>
<tr>
<td>REX.W + 90+<em>rd</em></td>
<td>XCHG <em>r64</em>, RAX</td>
<td>O</td>
<td>Valid</td>
<td>N.E.</td>
<td>Exchange RAX with <em>r64.</em></td></tr>
</table>

Reply:
As a side note, I noticed there is a test present to check if `xchg eax, eax` assembles to `90`. This does not roundtrip, but I kept the original behavior and test unchanged.